StandardLibraryEssentials/Templates/Introduction/task.md

```diff
@@ -0,0 +1,9 @@
+In this part of the course, you will learn about such a feature of the C++ language as templates.
+
+In essence, templates are a means to write generic code that can work with varying types. They allow you to write a single function or class that can adapt to different types, without the need to write separate functions or classes for each type.
+
+The following topics will be covered:
+ * Definition of templates for functions and classes 
+ * Type-inference, `auto` and `decltype` keywords, template argument deduction
+ * Type aliases: `using` and `typedef`
+ * Template specialization
```

Comment:
* Type inference: `auto` and `decltype`
* Template argument deduction
* Type aliases: `using` and `typedef`
* Template specialization